tests/verify_repl_deleteonfail.erl

```diff
@@ -0,0 +1,273 @@
+        [{riak_kv,
+            [{ttaaefs_scope, all},
+            {ttaaefs_localnval, 3},
+            {ttaaefs_remotenval, 1},
+            {ttaaefs_peerip, PeerIP},
+            {ttaaefs_peerport, PeerPort},
+            {ttaaefs_peerprotocol, pb},
+            {ttaaefs_allcheck, 0},
+            {ttaaefs_autocheck, 0},
+            {ttaaefs_daycheck, 0},
+            {ttaaefs_hourcheck, 0},
+            {ttaaefs_nocheck, 24},
+            {ttaaefs_maxresults, ?MAX_RESULTS},
+            {ttaaefs_queuename, LocalClusterName},
+            {ttaaefs_queuename_peer, RemoteClusterName},
+            {ttaaefs_logrepairs, true}]}]).
+
+confirm() ->
+    [ClusterA, ClusterB] =
+        rt:deploy_clusters([
+            {5, ?CFG_REPL("cluster_b:any", 3)},
+            {1, ?CFG_REPL("cluster_a:any", 1)}]),
+
+    lager:info("Discover Peer IP/ports and restart with peer config"),
+    reset_peer_config(ClusterA, ClusterB),
+
+    lists:foreach(
+        fun(N) -> rt:wait_until_ready(N), rt:wait_until_pingable(N) end,
+        ClusterA ++ ClusterB
+    ),
+    
+    rt:join_cluster(ClusterA),
+    rt:join_cluster(ClusterB),
+
+    ok = setup_fullsync_peer(ClusterA, hd(ClusterB)),
+    
+    ok = verify_repl_delete(ClusterA),
+    pass.
+
+
+verify_repl_delete(Nodes) ->
+    
+    lager:info("Commencing object load"),
+    KeyLoadFun =
+        fun(B, V) ->
+            fun(Node, KeyCount) ->
+                lager:info("Loading from key ~w on node ~w", [KeyCount, Node]),
+                KVs = 
+                    test_data(
+                        KeyCount + 1, KeyCount + ?NUM_KEYS_PERNODE, V),
+                ok = write_data(Node, B, KVs, []),
+                KeyCount + ?NUM_KEYS_PERNODE
+            end
+        end,
+
+    lists:foldl(KeyLoadFun(<<"B1">>, <<"v1">>), 1, Nodes),
+    lager:info("Loaded ~w objects", [?NUM_KEYS_PERNODE * length(Nodes)]),
+    wait_for_queues_to_drain(Nodes, cluster_b),
+    lists:foldl(KeyLoadFun(<<"B2">>, <<"v1">>), 1, Nodes),
+    lager:info("Loaded ~w objects", [?NUM_KEYS_PERNODE * length(Nodes)]),
+    wait_for_queues_to_drain(Nodes, cluster_b),
+    lists:foldl(KeyLoadFun(<<"B3">>, <<"v1">>), 1, Nodes),
+    lager:info("Loaded ~w objects", [?NUM_KEYS_PERNODE * length(Nodes)]),
+    wait_for_queues_to_drain(Nodes, cluster_b),
+    lists:foldl(KeyLoadFun(<<"B4">>, <<"v1">>), 1, Nodes),
+    lager:info("Loaded ~w objects", [?NUM_KEYS_PERNODE * length(Nodes)]),
+    wait_for_queues_to_drain(Nodes, cluster_b),
+   
+    
+    lager:info(
+        "Stopping a node - now tombstones will not be auto-reaped in A"),
+    FiddlingNode = hd(tl(Nodes)),
+    rt:stop_and_wait(FiddlingNode),
+    
+    lager:info("Delete all keys for B1 bucket"),
+    delete_data(
+        hd(Nodes), <<"B1">>, lists:seq(1, ?NUM_KEYS_PERNODE * length(Nodes))),
+    lager:info("Delete attempts completed"),
+    wait_for_queues_to_drain(Nodes -- [FiddlingNode], cluster_b),
+
```

Comment:
Strictly spoken, from the function name "wait_for_queues_to_drain", I cannot confirm here that the tombstones are deferred... Some observation of that fact after a certain time would make this test indeed show the point.

Most likely just a matter of some additional explanation on what is happening here